### ISSUE TITLE
angular sidenav responsive menu button toggle not working in safari until reload

### ISSUE THREAD

**Issue by alejandromartincruz** [angular/material2#14123]
#### Bug, feature request, or proposal:
When resizing from desktop to mobile size in Safari sidenav menu button do not displays until reload in mobile size.
If load on mobile size and reload to desktop size again with safari, then sidenav menu button won't dissapear until reload page in desktop size.

#### What is the expected behavior?
When resizing from desktop to mobile sidenav menu button should appear, and when resizing again to desktop should dissapear. In Chrome, and firefox it works, it only fails with safari.

#### What is the current behavior?
Button do not appear or dissapear when the media query breakpoints are reached only with safari browser.

#### What are the steps to reproduce?
$ ng new myproject
$ ng generate @angular/material:materialNav --name navigation
// remove <router-outlet></router-outlet> and add <navigation></navigation> to the app.component.html
$ ng serve

open safari with desktop width size and resize window to mobile size, button won't appear. Reload in mobile size and button will appear. Resize again to desktop size and button won't dissapear. Reload the page in desktop size and button will vanish.

#### What is the use-case or motivation for changing an existing behavior?
It works fine with any other browser, but it's not working with safari

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular CLI: 7.0.5
Node: 8.9.4
OS: darwin x64
Angular: 7.0.3
... animations, cdk, common, compiler, compiler-cli, core, forms
... http, language-service, material, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.10.3
@angular-devkit/build-angular     0.10.3
@angular-devkit/build-optimizer   0.10.3
@angular-devkit/build-webpack     0.10.3
@angular-devkit/core              7.0.3
@angular-devkit/schematics        7.0.5
@angular/cli                      7.0.5
@ngtools/webpack                  7.0.3
@schematics/angular               7.0.5
@schematics/update                0.10.5
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.19.1

**Comment by Trent-Matthias** [comment]
I believe this is the same issue i was having here https://github.com/angular/material2/issues/12479

The workaround i found was to add this at the end of the observable:
tap(() => this.changeDetectorRef.detectChanges())

This just tells angular to run change detection. I don't like that I need to do this only for safari, but it was the most clean way I could find to fix the bug on my end.

**Comment by alejandromartincruz** [comment]
I tried your solution, but now it doesn't work in Chrome or Firefox, neither Safari. Now in console I have an error:

`ERROR TypeError: Cannot read property 'detectChanges' of undefined
    at TapSubscriber._tapNext (navigation.component.ts:19)
    at TapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/tap.js.TapSubscriber._next (tap.js:45)
    at TapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (map.js:41)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (map.js:41)
    at MapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at DebounceTimeSubscriber.push../node_modules/rxjs/_esm5/internal/operators/debounceTime.js.DebounceTimeSubscriber.debouncedNext (debounceTime.js:48)
    at AsapAction.dispatchNext (debounceTime.js:62)
    at AsapAction.push../node_modules/rxjs/_esm5/internal/scheduler/AsyncAction.js.AsyncAction._execute (AsyncAction.js:63)`

In my code I have:

`[...]
import { Observable } from 'rxjs';
import { tap, map } from 'rxjs/operators';
[...]
isHandset$: Observable<boolean> = this.breakpointObserver.observe('(max-width: 768px)')
    .pipe(
      map(result => result.matches),
      tap(() => this.changeDetectorRef.detectChanges())
    );`

But this way don't seem to work. Moreover, I tried to find some information about tap, but couldn't find anything relevant.

**Comment by Trent-Matthias** [comment]
You need to inject the change detector like this in the constructor:` private changeDetectorRef: ChangeDetectorRef.`

**Comment by alejandromartincruz** [comment]
You are right, and I had also to import it from @angular/core. Now everything works again, but in Safari still no toggle button on resize, only with reload.

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
